### PR TITLE
Vite support

### PR DESCRIPTION
**Pull request by Z3d0X**
Adds a 2nd argument to `registerTheme`

```diff
- public function registerTheme(string $url): void
+ public function registerTheme(string $url, bool $vite = false): void
```

### Example
```php
Filament::serving(function () {
    Filament::registerTheme(
        'resources/css/filament.css',
        vite: true,
    );
});
```

### TODO
- [ ] update docs

### CODE REVIEW

**Comment by ArtMin96** [comment]
maybe
```php
if ($vite) {
    $this->themeUsesVite = true;
}
```

to

```php
 $this->themeUsesVite = $vite;
```

**Comment by danharrin** [comment]
Removed the extra param, now you can just do `Filament::registerTheme(app(Vite::class)('resources/css/filament.css')(`

**Comment by awcodes** [comment]
This look good to me. Works as expected from a Filament standpoint.

One thing I would add is to put autoprefixer in the npm install command and the postcss.config.js file in the docs, since it is still recommended by tailwind in their postcss docs.

**Review by zepfietje — CHANGES_REQUESTED** [review]
Haven't tested locally, but code looks good to me.

As @awcodes mentioned, autoprefixer should be added to the postcss config in the docs ([this is also how Laravel Jetstream sets this up](https://github.com/laravel/jetstream/blob/2.x/stubs/livewire/postcss.config.js)).

**Comment by pxlrbt** [comment]
Tested this with the demo and works.

**Review by pxlrbt — APPROVED** [review]

**Comment by danharrin** [comment]
Added autoprefixer, thanks

File: packages/admin/docs/07-appearance.md [hunk 1]
```diff
@@ -107,7 +107,37 @@ module.exports = {
...
+                // ...
+                'resources/css/filament.css',
+            ],
+            // ...
+        }),
+    ],
+})
+```
+
+And add Tailwind to the `postcss.config.js` file:
```
Comment (zepfietje) [inline packages/admin/docs/07-appearance.md:129]: Should we add autoprefixer to this sentence as well, @danharrin?

**Comment by pxlrbt** [comment]
@danharrin Seems like this causes issues with Laravel Mix. I didn't have that issue when testing with the default stylesheet, but when registering a theme `Filament::registerTheme(mix('css/filament.css'))`. 

Apparently `mix()` returns a HtmlString,

**Comment by danharrin** [comment]
Oh... right. We can fix that.

**Comment by danharrin** [comment]
hmmm

**Comment by danharrin** [comment]
how do we differentiate between mix and vite now...

**Comment by danharrin** [comment]
fixed

**Comment by danharrin** [comment]
I hate it but its fine

Screenshot 2022-07-17 at 22 42 08

**Comment by pxlrbt** [comment]
Looks nasty, but I guess it works. Had the same idea about checking for `<script`

**Pull request merged**